src/org/jitsi/impl/neomedia/MediaServiceImpl.java

```diff
@@ -1531,9 +1532,17 @@ public static boolean isJmfRegistryDisableLoad()
      */
     private static void postInitializeOnce(MediaServiceImpl mediaServiceImpl)
     {
-        new ZrtpFortunaEntropyGatherer(
-                mediaServiceImpl.getDeviceConfiguration())
-            .setEntropy();
+        /*
+         * Some SecureRandom() implementations like SHA1PRNG call
+         * /dev/random to seed themselves on first use.
+         * Call SecureRandom early to avoid blocking when establishing
+         * a connection for exemple.
+         */
+        logger.info("Warming up SecureRandom...");
+        SecureRandom rnd = new SecureRandom();
+        byte[] b = new byte[20];
+        rnd.nextBytes(b);
+        logger.info("Warming up SecureRandom finished.");
```

Comment:
I don't see how a sane compiler can decide to remove a function call like ```rnd.nextBytes(b);```.
how can it decide than there is no side affect when there actually are (SHA1PRNG seeding or NativePrng file opening)
Unless you know a buggy java version doing such harmful optimisation, i'm against defensive coding.